### PR TITLE
migratestatus tool: add url check for `-description` parameter

### PR DESCRIPTION
**Pull request by dhiller**
We encountered a 422 response when we set `-description` to a non url value:

```
{"component":"migratestatus","error":"status code 422 not one of [201], body: {\"message\":\"Validation Failed\",\"errors\":[{\"resource\":\"Status\",\"code\":\"custom\",\"field\":\"target_url\",\"message\":\"target_url must use http(s) scheme\"}],\"docu
mentation_url\":\"https://developer.github.com/v3/repos/statuses/#create-a-status\"}","file":"maintenance/migratestatus/migratestatus.go:144","func":"main.main","level":"fatal","msg":"Error during status migration","time":"2020-02-21T12:33:34+01:00"}
``` 

This commit adds a url check to the content of the description parameter to avoid failing on the server and fail early instead.

### CODE REVIEW

**Comment by dhiller** [comment]
Can do!

**Comment by dhiller** [comment]
@stevekuznetsov I've added the tests you requested.

**Comment by dhiller** [comment]
Thanks everyone, fixed the issues. PTAL.

File: maintenance/migratestatus/migratestatus.go [hunk 1]
```diff
@@ -75,6 +76,13 @@ func (o *options) Validate() error {
 		return errors.New("'--dest' is required unless using '--retire' mode.\n")
 	}
 
+	if o.descriptionURL != "" {
+		_, err := url.ParseRequestURI(o.descriptionURL)
+		if err != nil {
+			return errors.New("'--description' URL is not valid\n")
```
Comment (petr-muller) [inline maintenance/migratestatus/migratestatus.go]: nit: we should propagate the `err` out to better inform what went wrong

Comment (dhiller) [reply maintenance/migratestatus/migratestatus.go]: Good point.

**Comment by dhiller** [comment]
@michelle192837 can you please take another look?

**Pull request merged**